### PR TITLE
Modifies validator to flag unusual values for the DATS 'formats' field

### PR DESCRIPTION
**Pull request by cmadjar**
## Description

In order to harmonize DATS between datasets, a review of the available content for the distributions.formats field has been done and the following convention were determined:

file format should be capitalized without the dot preceding the format (exceptions being BigWig, NIfTI and RNA-Seq)
Examples of invalid format: tsv, .txt...
Examples of a valid format: TSV, TXT, MINC...

## Related issues

#491

Note: this is the same changes as in https://github.com/CONP-PCNO/conp-dataset/pull/498 but had to resubmit the PR as I pushed some dataset changes that should not have been in the PR modifying the validator and the tests...

### CODE REVIEW

**Comment by emmetaobrien** [comment]
I have had a reply from the lab who generate the openpain data, saying they will get back to us with number of subjects next week.

**Comment by cmadjar** [comment]
@zxenia the code is working. Ready for review.. Thanks!

FYI - The datasets that are failing the validation are either listed in #515 or will be fixed by #513 so independent of the code changes.

File: scripts/dats_validator/validator.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import logging
 import getopt
 from sys import argv
+from functools import reduce
```
Comment (zxenia) [inline scripts/dats_validator/validator.py:9]: unused import

File: tests/template.py [hunk 2]
```diff
@@ -52,12 +56,31 @@ def test_has_valid_dats(self, dataset):
...
 
+            # For crawled dataset, some tests should not be run as there is no way to
+            # automatically populate some of the fields
+            # For datasets crawled with Zenodo: check the formats extra property only
+            # For datasets crawled with OSF: skip validation of extra properties
+            is_osf_dataset = os.path.exists(os.path.join(dataset, '.conp-osf-crawler.json'))
+            is_zenodo_dataset = os.path.exists(os.path.join(dataset, '.conp-zenodo-crawler.json'))
+            is_valid, errors = validate_formats(json_obj) if is_zenodo_dataset else validate_non_schema_required(json_obj)
+            if not is_valid:
+                if not is_osf_dataset:
```
Comment (zxenia) [inline tests/template.py:74]: no need for two ifs, can be a one-line If statement with AND `if not is_valid and not is_osf_dataset:`

**Review by zxenia — APPROVED** [review]
Everything looks good! I have just a couple of minor comments not related to functionality.
After this, it can be merged.
Thanks @cmadjar

File: scripts/dats_validator/validator.py (hunk 1 above)
Comment (cmadjar) [reply scripts/dats_validator/validator.py:9]: nice catch!

File: tests/template.py (hunk 2 above)
Comment (cmadjar) [reply tests/template.py:74]: this is initially what I had but it did not work at the time. I re-changed it to see if it works now. Let's see what the tests say. Thanks!

**Comment by cmadjar** [comment]
@emmetaobrien if you could review/merge this PR now that it passes the checks?

Note: @zxenia already approved it but it does not show as she does not have write permission to the repo.

**Review by emmetaobrien — APPROVED** [review]
looks good to me.

**Pull request merged**